src/common/ad_mem.c

```diff
@@ -1731,14 +1907,31 @@ tx_complete(struct ad_tx *tx, int err)
 			pos = arena_locate_grp(group->gp_arena, group);
 			group->gp_unit_rsv--;
 			/* NB: reorder only does minimum amount of works most of the time */
+			arena = group->gp_arena;
 			arena_reorder_grp(group->gp_arena, group, pos, false);
+			node = &arena->ar_blob->bb_mh_nodes[arena->ar_df->ad_id];
```

Comment:
please use inline function to do this conversion.

Reply:
sure